### ISSUE TITLE
Feature request: character-based matcher

### ISSUE THREAD

**Issue by Matthalp** [clulab/processors#120]
This would be extremely useful for identifying entities such as phone numbers and e-mail addresses

**Comment by MihaiSurdeanu** [comment]
You can actually have subword patterns in Odin already, using the construct word = /REGEXHERE/

**Comment by myedibleenso** [comment]
@MihaiSurdeanu, I think @Matthalp is bringing up two things here: 

1) A matcher independent of tokenization that operates over a character sequence.  This means you wouldn't need to guess how the the text preprocessor and tokenizer will handle something like a phone number.  While on this surface this sounds like it's just a regex, it would be nice to have integration with Odin's other matchers.  Perhaps this matcher would simply operate over the original text and marry any match it to the respective tokens so that subsequent rules using dependencies or tokens could incorporate the output of the character-based rule.  

2) The ability to make a sub-token named captures.  This would also be useful in reach since many mutants and PTMs are represented as a character prefix (ex. `pPROTEIN-NAME` for a phosphorylation).

**Comment by hickst** [comment]
Yes, we have previously discussed what Marco is describing. I have really wanted this ability for a long time so that rules can be applied to the multiple "sub-token" grammars that grounding is currently hacking with fixed code. It seems to me that the power of rules at this lexico-syntactic level could really help us to:
1. increase the precision of our classifications
2. simplify tokenization
3. help with ER disambiguation problems
